### PR TITLE
Remove file with malformed name and expose offline.figToBuffer()

### PR DESCRIPTION
**Pull request by RyanCallahan312**
The [last commit](https://github.com/MetalBlueberry/go-plotly/commit/2e3261a23d0f6b44a049ff960c3c898227220cb4) to this repo prevented cloning the repo and any go gets due to a malformed file name the contained a checksum. I'm not sure why it was there because I can not find any reference to it. Maybe it was left out of the .gitignore? If this is the case I can add that onto this pr.

This pr also includes exporting the offline.figToBuffer() function. This function being exporting would enable using this library in use cases where writing to a file or serving an html file isn't an option. (Ex. WASM)

TL;DR
- removed file plot-schema?format=json&sha1=11b662302a42aa0698df091a9974ac8f6e1a2292 to allow go gets and repo cloning
- exported offline.figToBuffer() for custom use cases

### CODE REVIEW

**Comment by MetalBlueberry** [comment]
Hi @RyanCallahan312 
Thank you for pointing out that it was impossible to go get the repository, in fact, I committed by accident a file that was causing this issue. I think the issue happens because the file name is incompatible with zip file names. 

anyway, v0.3.1 is ready by just removing the file.

Talking about figToBuffer, I think it is not wise to use this method outside this scope. Let me elaborate on why.

figToBuffer is basically rendering the following template
``` html
	<head>
		<script src="https://cdn.plot.ly/plotly-1.58.4.min.js"></script>
	</head>
	</body>
		<div id="plot"></div>
	<script>
		data = JSON.parse('{{ . }}')
		Plotly.newPlot('plot', data);
	</script>
	<body>
```
If you have plans to render a figure this way and place it from wasm... It is not going to work. This will create multiple <head> tags and import multiple times plotly.js

This function exists because `ToHtml` generates a file containing this HTML code and `Serve` send the HTML directly to the browser.

good news are that you can still use this from wasm. You can use the json package to create a json string that plotly can understand. Then you can return this to javascript and do the magic there.

``` html
	<head>
		<script src="https://cdn.plot.ly/plotly-1.58.4.min.js"></script>
	</head>
	</body>
		<div id="plot"></div>
	<script>
(... stuff to initialize go wasam ...  register getFigure() string) 
		data = JSON.parse(getFigure())
		Plotly.newPlot('plot', data);
	</script>
	<body>
```

`getFigure` may look something like this, although I recommend properly handle errors instead of throwing panics.

```go
func getFigure() string {
	figBytes, err := json.Marshal(fig)
	if err != nil {
		panic(err)
	}
       return string(figBytes)
}
```

**Comment by RyanCallahan312** [comment]
I understand the reasoning on not exporting the figToBuffer() function, but I do believe that it could still be useful. Also thank you for your alternative and I entirely overlooked that option 😅.

I think my best argument for it would be an example of what my use case would be. I have a wasm project that is using figToBuffer and using the output as the srcDoc on an iframe which won't create any extra tags. It is essentially taking the generated template and rendering it.

 [Here's a gist](https://gist.github.com/RyanCallahan312/aa54d6790d63d8780a207ecec6697663) with my use case in action. Can't show too much because of academic honesty policy.

I believe that this use case would be entirely viable. In this case I could have used your alternative but I would have been going through the library to find the template then copy and pasting into my code and then generating the plotly stuff using your alternative. I think that would create a lot of unnecessary code and work duplication to get the same result that would have been achieved with just the one function call.

I might be missing something on this because I don't have too much experience with go yet and this is the first time I'm trying to contribute to a project, but I do believe this would be useful to my use case and other potential use cases that may not have thought of yet.

Ps. love the library! Very impressive ease of use and documentation to go with all of it!

**Comment by MetalBlueberry** [comment]
I've created a wasm example based on your needs =)
https://github.com/MetalBlueberry/go-plotly/tree/master/examples/wasm

If you are going to have more than one plot, this approach is easier and more efficient! 
I encourage you to add some sample data to the wasm example. now the map is empty. It can be your first contribution.

The main reason why I do not want to export figToBuffer is that the offline package is intended for prototyping. The methods ToHtml and Show are not designed to be efficient, but to be simple to use. In the end, you always use the json package to create a json object that will be interpreted by Plotly.js, but how you do this will change depending on your objective. In this case, you want to send data to plotly from wasm, but in my principal use case, I built this to be a web server for plotly figures.

**Comment by RyanCallahan312** [comment]
Ahh ok that makes more sense. Thank you for the explanation and taking the time to read my pr.

I believe we're in agreement that it should not be exported so I'm going to close the pr.
